### PR TITLE
Move away from eval

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY

use templating + literal_eval as a 'nicer' dynamic default config

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

configuration
##### ANSIBLE VERSION

```
2.4/2.5
```

### CODE REVIEW

File: lib/ansible/config/base.yml [hunk 1]
```diff
@@ -1255,7 +1255,7 @@ INVENTORY_ENABLED:
   type: list
 INVENTORY_IGNORE_EXTS:
   name: Inventory ignore extensions
-  default: eval(BLACKLIST_EXTS + ( '~', '.orig', '.ini', '.cfg', '.retry'))
+  default: "{{(BLACKLIST_EXTS + ( '~', '.orig', '.ini', '.cfg', '.retry'))|list}}"
```
Comment (abadger) [inline lib/ansible/config/base.yml]: Note: piping to list probably isn't needed (it's what literal_eval will do)

Comment (bcoca) [reply lib/ansible/config/base.yml]: would get tuple

**Comment by abadger** [comment]
shipit

**Comment by abadger** [comment]
Do we want to cherry-pick this so that 2.4.x has jinja2 in the config syntax?

**Comment by abadger** [comment]
Cherry-picked for 2.4.1.

**Pull request merged**